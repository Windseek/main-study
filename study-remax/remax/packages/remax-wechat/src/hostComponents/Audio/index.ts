import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

// 微信已不再维护
export const Audio: React.ComponentType = createHostComponent('audio');
