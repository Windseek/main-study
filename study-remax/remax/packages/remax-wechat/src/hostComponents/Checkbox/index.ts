import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps } from '../../types/component';

export interface CheckboxProps extends BaseProps {
  name?: string;
  /** checkbox标识，选中时触发checkbox-group的 change 事件，并携带 checkbox 的 value 1.0.0  */
  value?: string;
  /** 是否禁用 1.0.0  */
  disabled?: boolean;
  /** 当前是否选中，可用来设置默认选中 1.0.0  */
  checked?: boolean;
  /** (default: #09BB07) checkbox的颜色，同css的color 1.0.0  */
  color?: string;
}

/**
 * @see https://developers.weixin.qq.com/miniprogram/dev/component/checkbox.html
 */
export const Checkbox: React.ComponentType<CheckboxProps> = createHostComponent<CheckboxProps>('checkbox');

Checkbox.defaultProps = {
  disabled: false,
  checked: false,
  color: '#09bb07',
};
