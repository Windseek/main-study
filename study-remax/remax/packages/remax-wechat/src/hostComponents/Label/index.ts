import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps } from '../../types/component';

export interface LabelProps extends BaseProps {
  /** 绑定控件的 id 1.0.0 */
  for?: string;
}

/**
 * https://developers.weixin.qq.com/miniprogram/dev/component/label.html
 */
export const Label: React.ComponentType<LabelProps> = createHostComponent<LabelProps>('label');
