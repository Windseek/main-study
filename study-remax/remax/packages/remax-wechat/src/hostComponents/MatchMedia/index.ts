import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps } from '../../types/component';

export interface MatchMediaProps extends BaseProps {
  /** 页面最小宽度（ px 为单位） 2.11.1 */
  minWidth?: number;
  /** 页面最大宽度（ px 为单位） 2.11.1 */
  maxWidth?: number;
  /** 页面宽度（ px 为单位） 2.11.1 */
  width?: number;
  /** 页面最小高度（ px 为单位） 2.11.1 */
  minHeight?: number;
  /** 页面最大高度（ px 为单位） 2.11.1 */
  maxHeight?: number;
  /** 页面高度（ px 为单位） 2.11.1 */
  height?: number;
  /** 屏幕方向（ landscape 或 portrait ） 2.11.1 */
  orientation?: 'portrait' | 'landscape';
}
/**
 * @see https://developers.weixin.qq.com/miniprogram/dev/component/match-media.html
 */
export const MatchMedia: React.ComponentType<MatchMediaProps> = createHostComponent<MatchMediaProps>('match-media');
