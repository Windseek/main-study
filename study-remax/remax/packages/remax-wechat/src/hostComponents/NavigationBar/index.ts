import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps } from '../../types/component';

export interface NavigationBarProps extends BaseProps {
  /** 导航条标题 2.9.0 */
  title?: string;
  /** 是否在导航条显示 loading 加载提示 2.9.0 */
  loading?: boolean;
  /** 导航条前景颜色值，包括按钮、标题、状态栏的颜色，仅支持 #ffffff 和 #000000 2.9.0 */
  frontColor?: '#ffffff' | '#000000';
  /** 导航条背景颜色值，有效值为十六进制颜色 2.9.0 */
  backgroundColor?: string;
  /** 改变导航栏颜色时的动画时长，默认为 0 （即没有动画效果） 2.9.0 */
  colorAnimationDuration?: number;
  /** 改变导航栏颜色时的动画方式，支持 linear 、 easeIn 、 easeOut 和 easeInOut 2.9.0 */
  colorAnimationTimingFunc?: 'linear' | 'easeIn' | 'easeOut' | 'easeInOut';
}
/**
 * 页面导航条配置节点，用于指定导航栏的一些属性。只能是 page-meta 组件内的第一个节点，需要配合它一同使用。
 * @see https://developers.weixin.qq.com/miniprogram/dev/component/navigation-bar.html
 */
export const NavigationBar: React.ComponentType<NavigationBarProps> = createHostComponent<NavigationBarProps>(
  'navigation-bar'
);

NavigationBar.defaultProps = {
  loading: false,
  colorAnimationDuration: 0,
  colorAnimationTimingFunc: 'linear',
};
