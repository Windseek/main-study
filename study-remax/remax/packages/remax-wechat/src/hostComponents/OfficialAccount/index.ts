import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps, GenericEvent } from '../../types/component';

export interface OfficialAccountProps extends BaseProps {
  /** 组件加载成功时触发 */
  onLoad?: (event: GenericEvent) => any;
  /** 组件加载失败时触发 */
  onError?: (event: GenericEvent) => any;
}

/**
 * https://developers.weixin.qq.com/miniprogram/dev/component/official-account.html
 */
export const OfficialAccount: React.ComponentType<OfficialAccountProps> = createHostComponent<OfficialAccountProps>(
  'official-account'
);
