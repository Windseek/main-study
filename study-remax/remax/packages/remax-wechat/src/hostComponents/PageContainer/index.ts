import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps, GenericEvent } from '../../types/component';

export interface PageContainerProps extends BaseProps {
  /** 是否显示容器组件 2.16.0 */
  show?: boolean;
  /** 动画时长，单位毫秒 2.16.0 */
  duration?: number;
  /** z-index 层级 2.16.0 */
  zIndex?: number;
  /** 是否显示遮罩层 2.16.0 */
  overlay?: boolean;
  /** 弹出位置，可选值为 top bottom right center 2.16.0 */
  position?: 'top' | 'bottom' | 'right' | 'center';
  /** 是否显示圆角 2.16.0 */
  round?: boolean;
  /** 是否在下滑一段距离后关闭 2.16.0 */
  closeOnSlideDown?: boolean;
  /** 自定义遮罩层样式 2.16.0 */
  overlayStyle?: string;
  /** 自定义弹出层样式 2.16.0 */
  customStyle?: string;
  /** 进入前触发 2.16.0 */
  onBeforeEnter?: (event: GenericEvent) => any;
  /** 进入中触发 2.16.0 */
  onEnter?: (event: GenericEvent) => any;
  /** 进入后触发 2.16.0 */
  onAfterEnter?: (event: GenericEvent) => any;
  /** 离开前触发 2.16.0 */
  onBeforeLeave?: (event: GenericEvent) => any;
  /** 离开中触发 2.16.0 */
  onLeave?: (event: GenericEvent) => any;
  /** 离开后触发 2.16.0 */
  onAfterLeave?: (event: GenericEvent) => any;
  /** 点击遮罩层触发 2.16.0 */
  onClickOverlay?: (event: GenericEvent) => any;
}
/**
 * @see https://developers.weixin.qq.com/miniprogram/dev/component/page-container.html
 */
export const PageContainer: React.ComponentType<PageContainerProps> = createHostComponent<PageContainerProps>(
  'page-container'
);

PageContainer.defaultProps = {
  show: false,
  duration: 300,
  zIndex: 100,
  overlay: true,
  position: 'bottom',
  round: false,
  closeOnSlideDown: false,
};
