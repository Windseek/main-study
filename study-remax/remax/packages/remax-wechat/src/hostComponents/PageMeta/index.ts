import * as React from 'react';
import { createHostComponent } from '@remax/runtime';
import { BaseProps, GenericEvent } from '../../types/component';

export interface PageMetaProps extends BaseProps {
  /** 下拉背景字体、loading 图的样式，仅支持 dark 和 light 2.9.0 */
  backgroundTextStyle?: 'dark' | 'light';
  /** 窗口的背景色，必须为十六进制颜色值 2.9.0 */
  backgroundColor?: string;
  /** 顶部窗口的背景色，必须为十六进制颜色值，仅 iOS 支持 2.9.0 */
  backgroundColorTop?: string;
  /** 底部窗口的背景色，必须为十六进制颜色值，仅 iOS 支持 2.9.0 */
  backgroundColorBottom?: string;
  /** 页面内容的背景色，用于页面中的空白部分和页面大小变化 resize 动画期间的临时空闲区域 2.12.1 */
  rootBackgroundColor?: string;
  /** 滚动位置，可以使用 px 或者 rpx 为单位，在被设置时，页面会滚动到对应位置 2.9.0 */
  scrollTop?: string;
  /** 滚动动画时长 2.9.0 */
  scrollDuration?: number;
  /** 页面根节点样式，页面根节点是所有页面节点的祖先节点，相当于 HTML 中的 body 节点 2.9.0 */
  pageStyle?: string;
  /** 页面 page 的字体大小，可以设置为 system ，表示使用当前用户设置的微信字体大小 2.11.0 */
  pageFontSize?: string;
  /** 页面的根字体大小，页面中的所有 rem 单位，将使用这个字体大小作为参考值，即 1rem 等于这个字体大小；自小程序版本 2.11.0 起，也可以设置为 system 2.9.0 */
  rootFontSize?: string;
  /** 页面的方向，可为 auto portrait 或 landscape 2.12.0 */
  pageOrientation?: 'auto' | 'portrait' | 'landscape';
  /** 页面尺寸变化时会触发 resize 事件， event.detail = { size: { windowWidth, windowHeight } } 2.9.0 */
  onResize?: (event: GenericEvent) => any;
  /** 页面滚动时会触发 scroll 事件， event.detail = { scrollTop } 2.9.0 */
  onScroll?: (event: GenericEvent) => any;
  /** 如果通过改变 scroll-top 属性来使页面滚动，页面滚动结束后会触发 scrolldone 事件 2.9.0 */
  onScrollDone?: (event: GenericEvent) => any;
}
/**
 * 页面属性配置节点，用于指定页面的一些属性、监听页面事件。只能是页面内的第一个节点。可以配合 navigation-bar 组件一同使用。
 * @see https://developers.weixin.qq.com/miniprogram/dev/component/page-meta.html
 */
export const PageMeta: React.ComponentType<PageMetaProps> = createHostComponent<PageMetaProps>('page-meta');

PageMeta.defaultProps = {
  scrollDuration: 300,
};
