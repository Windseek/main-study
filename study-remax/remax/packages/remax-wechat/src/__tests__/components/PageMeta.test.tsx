import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import { PageMeta } from '../../hostComponents';

describe('PageMeta', () => {
  it('render correctly', () => {
    const testRenderer = TestRenderer.create(<PageMeta />);

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
