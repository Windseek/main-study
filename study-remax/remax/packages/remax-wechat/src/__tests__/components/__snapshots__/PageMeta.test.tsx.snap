// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMeta render correctly 1`] = `
<page-meta
  scrollDuration={300}
/>
`;
